### PR TITLE
Remove findDOMNode and defaultProps from suppressed warnings

### PR DESCRIPTION
**Pull request by ravicious**
Now that #46590 has been merged, we can remove `findDOMNode` from suppressed warnings. This should help with making sure that we don't add new code which uses this deprecated API.

The `defaultProps` doesn't need to be in the suppressed warnings as well. I'm not sure at which point it stopped being needed, but it's likely that we had a function component that was using `defaultProps`. In React 19, function components will no longer be able to use default props, but class components should remain unchanged in that regard. The full message for that warning was:

> Warning: Button: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.

### CODE REVIEW

**Review by gzdunek — APPROVED** [review]

File: web/packages/build/jest/setupTests.ts [hunk 1]
```diff
@@ -53,24 +53,16 @@ const rootDir = path.join(__dirname, '..', '..', '..', '..');
 // If the test is expected to use either of those console functions, follow the advice from the
 // error message.
 const failOnConsoleIgnoreList = new Set([
+  ...entFailOnConsoleIgnoreList,
   'web/packages/design/src/utils/match/matchers.test.ts',
   'web/packages/shared/components/TextEditor/TextEditor.test.tsx',
   'web/packages/teleport/src/components/BannerList/useAlerts.test.tsx',
-  'web/packages/teleport/src/Navigation/NavigationItem.test.tsx',
```
Comment (ravicious) [inline web/packages/build/jest/setupTests.ts:59]: No longer seems to produce warnings.

File: web/packages/build/jest/setupTests.ts [hunk 2]
```diff
@@ -53,24 +53,16 @@ const rootDir = path.join(__dirname, '..', '..', '..', '..');
...
-  // As of the parent commit (708dac8e0d0), the tests below are flakes.
-  // https://github.com/gravitational/teleport/pull/41252#discussion_r1595036569
-  'web/packages/teleport/src/Console/DocumentNodes/DocumentNodes.story.test.tsx',
-  'web/packages/teleport/src/Recordings/Recordings.story.test.tsx',
-  'web/packages/teleport/src/Audit/Audit.story.test.tsx',
```
Comment (ravicious) [inline web/packages/build/jest/setupTests.ts:64]: These three no longer exist.

File: web/packages/build/jest/setupTests.ts [hunk 3]
```diff
@@ -53,24 +53,16 @@ const rootDir = path.join(__dirname, '..', '..', '..', '..');
...
-  'web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.test.tsx',
```
Comment (ravicious) [inline web/packages/build/jest/setupTests.ts:65]: No longer seems to produce warnings. Originally it produced this:

```
FAIL web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.test.tsx
  ● test AutoDeploy.tsx › deploy hint states

    Expected test not to call console.error().

    If the error is expected, test for it explicitly by mocking it out using jest.spyOn(console, 'error').mockImplementation() and test that the warning occurs.

    Error: Error: connect ECONNREFUSED 127.0.0.1:80

      41 |
      42 | export const SelectSecurityGroups = ({
    > 43 |   selectedSecurityGroups,
         |   ^
      44 |   setSelectedSecurityGroups,
      45 |   dbMeta,
      46 |   emitErrorEvent,

      at Object.dispatchError (node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:63:19)
      at processTicksAndRejections (node:internal/process/task_queues:82:21) {
        type: 'XMLHttpRequest'
      }
          at console.captureMessage [as error] (/__w/teleport.e/teleport.e/node_modules/jest-fail-on-console/index.js:78:25)
          at VirtualConsole.<anonymous> (/__w/teleport.e/teleport.e/node_modules/jest-environment-jsdom/build/index.js:63:23)
          at VirtualConsole.emit (node:events:519:28)
          at Object.dispatchError (/__w/teleport.e/teleport.e/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:66:53)
          at Request.<anonymous> (/__w/teleport.e/teleport.e/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:655:18)
          at Request.emit (node:events:531:35)
          at ClientRequest.<anonymous> (/__w/teleport.e/teleport.e/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:121:14)
          at ClientRequest.emit (node:events:519:28)
          at Socket.socketErrorListener (node:_http_client:500:9)
          at Socket.emit (node:events:519:28)
          at emitErrorNT (node:internal/streams/destroy:169:8)
          at emitErrorCloseNT (node:internal/streams/destroy:128:3)
          at processTicksAndRejections (node:internal/process/task_queues:82:21)
```

See [the Slack thread](https://gravitational.slack.com/archives/C02DQ1C2BMW/p1715790851948379) and https://github.com/gravitational/teleport/pull/41551 which added this line to the ignore list.

**Review by ravicious — COMMENTED** [review]
I remove a couple of tests from the ignore list. For the ones that still exist, I ran `pnpm test` for each of them 100 times.

**Review by zmb3 — APPROVED** [review]

**Review by gzdunek — APPROVED** [review]

**Pull request merged**